integration_tests/tests/signIn.ts

```diff
@@ -7,10 +7,10 @@ export const signIn = (
   user?: Partial<ApprovedPremisesUser>,
 ) => {
   cy.task('stubSignIn')
-  const permissions = makeArrayOfType<ApprovedPremisesUserRole>(role).reduce(
+  const rolesPermissions = makeArrayOfType<ApprovedPremisesUserRole>(role).reduce(
     (permissionList, userRole: ApprovedPremisesUserRole) => [...permissionList, ...roleToPermissions(userRole)],
     [] as Array<ApprovedPremisesUserPermission>,
   )
-  cy.task('stubAuthUser', { role, permissions, ...user })
+  cy.task('stubAuthUser', { role, ...user, permissions: [...rolesPermissions, ...(user?.permissions || [])] })
```

Comment:
This is a subtle change, which means `signIn` now considers any `permissions` as _additional_ to the ones that exist for the role. This ensures no test lacks base permissions for the given role, but allows extending those permissions, which is useful for experimental permissions.

Reply:
That's correct -- thinking about it, I think it was only really a problem while I had both old and new flows running alongside each other. Ultimately I do think it's a more logical approach -- without it, some of the tests were using permissions that were not representative of real world scenarios at all, which, I think, would be more prone to errors... I could be swayed!